### PR TITLE
Add Constraint BBE

### PR DESCRIPTION
**Pull request by TharmiganK**
> Fixes : [`Add a BBE for constraint module #3208`](https://github.com/ballerina-platform/ballerina-standard-library/issues/3208)

### CODE REVIEW

File: examples/constraint-validations/constraint_validations.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Hello world main
```
Comment (chamil321) [inline examples/constraint-validations/constraint_validations.md:1]: Lets change this

Comment (TharmiganK) [inline examples/constraint-validations/constraint_validations.md:1]:
```suggestion
# Constraint validations
```

Comment (TharmiganK) [reply examples/constraint-validations/constraint_validations.md]: Fixed vis [732d286](https://github.com/ballerina-platform/ballerina-distribution/pull/3060/commits/732d2860a98863b093c74da6f1ec67631c29e11d)

File: examples/constraint-validations/constraint_validations.bal [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import ballerina/constraint;
+
+// Constraint on `int` type.
```
Comment (praneesha) [inline examples/constraint-validations/constraint_validations.bal:3]:
```suggestion
// Constraint on the `int` type.
```

File: examples/constraint-validations/constraint_validations.bal [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+@constraint:Int {
+    minValue: 18
+}
+type Age int;
+
+type Student record {|
+    // Constraint on `string` typed record field.
```
Comment (praneesha) [inline examples/constraint-validations/constraint_validations.bal:10]:
```suggestion
    // Constraint on the `string`-typed record field.
```

File: examples/constraint-validations/constraint_validations.bal [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    @constraint:String {
+        length: 7
+    }
+    string id;
+    string name;
+    // Constrainted type used as a record field.
+    Age age;
+    // Constraint on `string[]` typed record field;
```
Comment (praneesha) [inline examples/constraint-validations/constraint_validations.bal:18]:
```suggestion
    // Constraint on the `string[]`-typed record field.
```

File: examples/constraint-validations/constraint_validations.bal [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+|};
+
+public function main() returns error? {
+    Student student = {
+        id: "200146B",
+        name: "David John",
+        age: 25,
+        subjects: ["Maths", "Science", "English"]
+    };
+    // To validate the constraints on `Student` record, the `validate` function should be
```
Comment (praneesha) [inline examples/constraint-validations/constraint_validations.bal:33]:
```suggestion
    // To validate the constraints on the `Student` record, the `validate` function should be 
```

File: examples/constraint-validations/constraint_validations.bal [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+public function main() returns error? {
+    Student student = {
+        id: "200146B",
+        name: "David John",
+        age: 25,
+        subjects: ["Maths", "Science", "English"]
+    };
+    // To validate the constraints on `Student` record, the `validate` function should be 
+    // called explicitly. If the validation is successful then this function returns the type
```
Comment (praneesha) [inline examples/constraint-validations/constraint_validations.bal:34]:
```suggestion
    // called explicitly. If the validation is successful, then, this function returns the type 
```

File: examples/constraint-validations/constraint_validations.bal [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+public function main() returns error? {
+    Student student = {
+        id: "200146B",
+        name: "David John",
+        age: 25,
+        subjects: ["Maths", "Science", "English"]
+    };
+    // To validate the constraints on `Student` record, the `validate` function should be 
+    // called explicitly. If the validation is successful then this function returns the type 
+    // descriptor of the value which is validated.
```
Comment (praneesha) [inline examples/constraint-validations/constraint_validations.bal:35]:
```suggestion
    // descriptor of the value that is validated.
```

File: examples/constraint-validations/constraint_validations.bal [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+    student = check constraint:validate(student);
+
+    // Set the student's age to 17 which will violate the `minValue` constraint on `Age`.
```
Comment (praneesha) [inline examples/constraint-validations/constraint_validations.bal:38]:
```suggestion
    // Set the student's age to 17, which will violate the `minValue` constraint on `Age`.
```

File: examples/constraint-validations/constraint_validations.description [hunk 9]
```diff
@@ -0,0 +1,6 @@
+// Validating user input is a common requirement in most applications. This can prevent user entry 
+// errors before the app attempts to process the data. The `constraint` library provides such 
+// validations for the following Ballerina types : `int`, `float`, `decimal`, `string` and `anydata[]`.
```
Comment (praneesha) [inline examples/constraint-validations/constraint_validations.description:3]:
```suggestion
// validations for the following Ballerina types: `int`, `float`, `decimal`, `string`, and `anydata[]`.
```

File: examples/constraint-validations/constraint_validations.md [hunk 10]
```diff
@@ -0,0 +1,11 @@
+# Constraint validations
+
+Validating user input is a common requirement in most applications. 
+This can prevent user entry errors before the app attempts to process the data. 
+The `constraint` library provides such validations for the following Ballerina types : `int`, `float`, `decimal`, `string` and `anydata[]`.
```
Comment (praneesha) [inline examples/constraint-validations/constraint_validations.md:5]:
```suggestion
The `constraint` library provides such validations for the following Ballerina types: `int`, `float`, `decimal`, `string`, and `anydata[]`.
```

**Review by chamil321 — APPROVED** [review]

**Pull request merged**